plugin/src/main/java/org/opensearch/ml/action/IndexInsight/CreateIndexInsightContainerTransportAction.java

```diff
@@ -0,0 +1,228 @@
+import org.opensearch.ml.common.transport.indexInsight.MLIndexInsightContainerCreateAction;
+import org.opensearch.ml.common.transport.indexInsight.MLIndexInsightContainerCreateRequest;
+import org.opensearch.ml.engine.indices.MLIndicesHandler;
+import org.opensearch.ml.utils.TenantAwareHelper;
+import org.opensearch.remote.metadata.client.GetDataObjectRequest;
+import org.opensearch.remote.metadata.client.PutDataObjectRequest;
+import org.opensearch.remote.metadata.client.SdkClient;
+import org.opensearch.remote.metadata.common.SdkClientUtils;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+import org.opensearch.transport.client.Client;
+
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * This action will put a container object into system index, then
+ * Create an index to store index insight using the provided index name if not exists.
+ */
+
+@Log4j2
+public class CreateIndexInsightContainerTransportAction extends HandledTransportAction<ActionRequest, AcknowledgedResponse> {
+    private Client client;
+    private final SdkClient sdkClient;
+    private NamedXContentRegistry xContentRegistry;
+    private MLFeatureEnabledSetting mlFeatureEnabledSetting;
+    private final MLIndicesHandler mlIndicesHandler;
+
+    @Inject
+    public CreateIndexInsightContainerTransportAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        NamedXContentRegistry xContentRegistry,
+        MLFeatureEnabledSetting mlFeatureEnabledSetting,
+        Client client,
+        SdkClient sdkClient,
+        MLIndicesHandler mlIndicesHandler
+    ) {
+        super(MLIndexInsightContainerCreateAction.NAME, transportService, actionFilters, MLIndexInsightContainerCreateRequest::new);
+        this.client = client;
+
+        this.xContentRegistry = xContentRegistry;
+        this.mlFeatureEnabledSetting = mlFeatureEnabledSetting;
+        this.sdkClient = sdkClient;
+        this.mlIndicesHandler = mlIndicesHandler;
+    }
+
+    @Override
+    protected void doExecute(Task task, ActionRequest request, ActionListener<AcknowledgedResponse> listener) {
+        MLIndexInsightContainerCreateRequest mlIndexInsightContainerCreateRequest = MLIndexInsightContainerCreateRequest
+            .fromActionRequest(request);
+        if (!TenantAwareHelper.validateTenantId(mlFeatureEnabledSetting, mlIndexInsightContainerCreateRequest.getTenantId(), listener)) {
+            return;
+        }
+
+        String tenantId = mlIndexInsightContainerCreateRequest.getTenantId();
+        IndexInsightContainer indexInsightContainer = IndexInsightContainer
+            .builder()
+            .indexName(mlIndexInsightContainerCreateRequest.getIndexName())
+            .tenantId(tenantId)
+            .build();
+        // The container is a doc in system index, and it defines where we store the index insight. The index insight is an user index
+        // inside user's cluster.
```

Comment:
This comment is confusing. I think you mean "The index insight is for an user index inside user's cluster".

Reply:
Already change it.